plugins/inputs/icinga2/icinga2.go

```diff
@@ -123,39 +195,74 @@ func (i *Icinga2) Gather(acc telegraf.Accumulator) error {
 		i.client = client
 	}
 
-	requestURL := "%s/v1/objects/%s?attrs=name&attrs=display_name&attrs=state&attrs=check_command"
-
-	// Note: attrs=host_name is only valid for 'services' requests, using check.Attrs.HostName for the host
-	//       'hosts' requests will need to use attrs=name only, using check.Attrs.Name for the host
-	if i.ObjectType == "services" {
-		requestURL += "&attrs=host_name"
+	// For backward config compatibility
+	if i.ObjectType != "" {
+		i.Objects = []string{i.ObjectType}
 	}
```

Comment:
This is something to be placed in the `func (i *Icinga2) Init()` method, it doesn't need to be executed every gather cycle.

Reply:
@Hipska I'm not sure how to access the toml config to check for `i.ObjectType` from the `Init()` method. Is there any input that is doing this kind of validation?

Reply:
Almost all of them? See https://github.com/influxdata/telegraf/blob/410226051daa56ced3acde1df97d5e147299ce7f/plugins/inputs/example/example.go#L44-L63

Reply:
Sorry, I've overlooked that.